flink-runtime/src/main/java/org/apache/flink/runtime/zookeeper/ZooKeeperStateHandleStore.java

```diff
@@ -346,11 +346,7 @@ public int exists(String pathInZooKeeper) throws Exception {
 			} else {
 				// Initial cVersion (number of changes to the children of this node)
 				int initialCVersion = stat.getCversion();
-
-				List<String> children = ZKPaths.getSortedChildren(
-						client.getZookeeperClient().getZooKeeper(),
-						ZKPaths.fixForNamespace(client.getNamespace(), "/"));
-
+				List<String> children = client.getZookeeperClient().getZooKeeper().getChildren(ZKPaths.fixForNamespace(client.getNamespace(), "/"), false);
```

Comment:
I think this alone does not work: The JavaDocs of `ZooKeeper#getChildren` say

> The list of children returned is not sorted and no guarantee is provided as to its natural or lexical order.

Thus, I assume that it is not safe to simply return the list of children without any further processing.